clients/pkg/promtail/targets/gcplog/pull_target.go

```diff
@@ -0,0 +1,164 @@
+package gcplog
+
+import (
+	"context"
+	"sync"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/target"
+)
+
+// PullTarget represents the target specific to GCP project, with a pull subscription type.
+// It collects logs from GCP and push it to Loki.
+// nolint:revive
+type PullTarget struct {
```

Comment:
Now `PullTarget` doesn't have to be public correct?.

Reply:
Yup, since the actual type returned by the public `NewGCPLogTarget` func is `gcplog.Target`, we can keep the actual target types package private.